### PR TITLE
StateSeparator Tests

### PR DESCRIPTION
**Pull request by gwwatkin**
- [x] `trace_dict_state`
 - [x] `trace_to_density_op`
 - [x] `separate`
 - [x] `get_separable_qubits`

Test required to work on the functionality for #138

### CODE REVIEW

File: src/lsqecc/simulation/qiskit_opflow_utils.py [hunk 1]
```diff
@@ -85,5 +84,15 @@ def get_separable_qubits(dict_state: qkop.DictStateFn) -> Dict[int, qkop.DictSta
...
+
+
+def to_dict_fn(vector_state: qkop.OperatorBase) -> qkop.DictStateFn:
+    # Switch two identical cases to keep mypy happy
+    if isinstance(vector_state, qkop.VectorStateFn):
+        return cast(qkop.DictStateFn, vector_state.to_dict_fn())
+    elif isinstance(vector_state, qkop.SparseVectorStateFn):
+        return cast(qkop.DictStateFn, vector_state.to_dict_fn())
+    else:
+        raise NotImplementedError(f"Conversion to_dict_fn of {repr(type(vector_state))}")
```
Comment (alexnguyenn) [inline src/lsqecc/simulation/qiskit_opflow_utils.py:98]: codecov is asking for a test case here 😅 (L98).

**Review by alexnguyenn — APPROVED** [review]
Nice to see more tests added. I saw you did run into some troubles with mypy - personally I don't think typing is that necessary for tests? 

Can't comment on the correctness so we need @isolatedinformation's comments.

**Comment by gwwatkin** [comment]
@alexnguyenn these tests juggle a lot of similar but not the same types like `StateVector`, vector `StateFn`s and numpy arrays so I ended up writing annotations to make sense of what was going on, but generally I would say typing isn't necessary for tests

**Comment by gwwatkin** [comment]
@isolatedinformation  FYI

**Pull request merged**